.github/workflows/mac-test.yml

```diff
@@ -13,16 +13,46 @@ jobs:
       max-parallel: 4
       matrix:
         os: [macos-latest]
-        python-version: ['3.10', '3.11', '3.12']
+        python-version:
+          - '3.10'
+          - '3.11'
+          - '3.12'
+
+    # this is needed for conda environments to activate automatically
+    defaults:
+      run:
+        shell: bash -el {0}
+
     steps:
     - uses: actions/checkout@v1
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
+
+    - name: Cache conda packages
+      uses: actions/cache@v4
+      env:
+        # increment to reset cache
+        CACHE_NUMBER: 0
+      with:
+        path: ~/conda_pkgs_dir
+        key: ${{ runner.os }}-conda-${{ matrix.python-version}}-${{ env.CACHE_NUMBER }}
+
+    - name: Configure conda
+      uses: conda-incubator/setup-miniconda@v3
       with:
+        activate-environment: test
+        channels: conda-forge
+        miniforge-version: latest
         python-version: ${{ matrix.python-version }}
-    - run: |
-        brew install fftw openssl gsl
-        pip install --upgrade pip setuptools "tox<4.0.0"
-    - name: run basic pycbc test suite
+
+    - name: Conda info
+      run: conda info --all
+
+    - name: Install tox
+      run: |
+        conda install \
+            pip \
+            setuptools \
+            "tox<4.0.0"
```

Comment:
Is this restriction understood?

Reply:
The reason was discussed here https://github.com/gwastro/pycbc/pull/4208 ... It may no longer apply.